substrate/frame/staking/ah-client/src/lib.rs

```diff
@@ -0,0 +1,318 @@
+///
+/// The codec index must
+/// correspond to the index of `pallet-staking-rc-client` in the `construct_runtime` of AssetHub.
+#[derive(Encode, Decode)]
+enum AssetHubRuntimePallets {
+	#[codec(index = 50)]
+	RcClient(StakingCalls),
+}
+
+/// Call encoding for the calls needed from the Broker pallet.
+#[derive(Encode, Decode)]
+enum StakingCalls {
+	#[codec(index = 1)]
+	StartSession(SessionIndex),
+	#[codec(index = 2)]
+	EndSession(SessionIndex, Vec<(AccountId32, u32)>),
+	#[codec(index = 3)]
+	NewOffence(SessionIndex, Vec<Offence>),
+}
+
+// Based on [`sp_staking::offence::OffenceDetails`].
+#[derive(Encode, Decode)]
+struct Offence {
+	offender: AccountId32,
+	reporters: Vec<AccountId32>,
+	slash_fraction: Perbill,
+}
+
+#[frame_support::pallet(dev_mode)]
+pub mod pallet {
+	use crate::*;
+	use alloc::vec;
+	use core::result;
+	use frame_system::pallet_prelude::*;
+	use pallet_session::historical;
+	use pallet_staking::ExposureOf;
+	use polkadot_primitives::Id as ParaId;
+	use polkadot_runtime_parachains::origin::{ensure_parachain, Origin};
+	use sp_runtime::Perbill;
+	use sp_staking::{offence::OnOffenceHandler, SessionIndex};
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	// TODO: should contain some initial state, otherwise starting from genesis won't work
+	#[pallet::storage]
+	pub type ValidatorSet<T: Config> = StorageValue<
+		_,
+		Option<Vec<(T::AccountId, Exposure<T::AccountId, BalanceOf<T>>)>>,
+		ValueQuery,
+	>;
+
+	/// How many blocks each validator has authored during the current session.
+	#[pallet::storage]
+	pub type BlockAuthors<T: Config> = StorageMap<_, Twox64Concat, AccountId32, u32, ValueQuery>;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeOrigin: From<<Self as frame_system::Config>::RuntimeOrigin>
+			+ Into<result::Result<Origin, <Self as Config>::RuntimeOrigin>>;
+		/// Just the `Currency::Balance` type; we have this item to allow us to constrain it to
+		/// `From<u64>`.
+		type CurrencyBalance: sp_runtime::traits::AtLeast32BitUnsigned
+			+ codec::FullCodec
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ core::fmt::Debug
+			+ Default
+			+ From<u64>
+			+ TypeInfo
+			+ Send
+			+ Sync
+			+ MaxEncodedLen;
+		/// The ParaId of the AssetHub.
+		#[pallet::constant]
+		type AssetHubId: Get<u32>;
+		/// The XCM sender.
+		type SendXcm: SendXcm;
+		/// Maximum weight for any XCM transact call that should be executed on AssetHub.
+		type MaxXcmTransactWeight: Get<Weight>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The paraid making the call is not AssetHub.
+		NotAssetHub,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		// #[pallet::weight(T::WeightInfo::new_validators())] // TODO
+		pub fn new_validators(
+			origin: OriginFor<T>,
+			new_validator_set: Vec<(T::AccountId, Exposure<T::AccountId, BalanceOf<T>>)>,
```

Comment:
This should only receive vec of validator ids. 
```suggestion
			new_validator_set: Vec<T::AccountId>,
```